app/models/user.rb

```diff
@@ -9,4 +9,30 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
```

Comment:
Rails associations are really powerful and you should take advantage of them. They are more efficient than enumerable methods (querying the DB with them is not a good practice).
For instance, you can create a few to check the status of the friendship, instead of building methods for that (associations are Rails built-in methods already!)